client/lib/plans/features-list.js

```diff
@@ -357,7 +357,7 @@ export const FEATURES_LIST = {
 
 	[ constants.FEATURE_JETPACK_ADVANCED ]: {
 		getSlug: () => constants.FEATURE_JETPACK_ADVANCED,
-		getTitle: () => i18n.translate( 'Jetpack Advanced Features' ),
+		getTitle: () => i18n.translate( 'Jetpack advanced features' ),
```

Comment:
:warning: This change will be queued for retranslation. We'll display the English text until then.